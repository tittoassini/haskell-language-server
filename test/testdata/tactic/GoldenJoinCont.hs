type Cont r a = ((a -> r) -> r)

joinCont :: Show a => (a -> c) -> (b -> c) -> Either a b -> (c -> d) -> d
joinCont = _
