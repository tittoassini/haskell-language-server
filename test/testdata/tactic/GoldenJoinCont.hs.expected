type Cont r a = ((a -> r) -> r)

joinCont :: Show a => (a -> c) -> (b -> c) -> Either a b -> (c -> d) -> d
joinCont = (\ fac fbc eab fcd
   -> case eab of
        (Left a) -> fcd (fac a)
        (Right b) -> fcd (fbc b))
